text_formatter.go

```diff
@@ -109,19 +140,16 @@ func (f *TextFormatter) Format(entry *Entry) ([]byte, error) {
 	return b.Bytes(), nil
 }
 
-func (f *TextFormatter) printColored(b *bytes.Buffer, entry *Entry, keys []string, timestampFormat string) {
-	var levelColor int
-	switch entry.Level {
-	case DebugLevel:
-		levelColor = gray
-	case WarnLevel:
-		levelColor = yellow
-	case ErrorLevel, FatalLevel, PanicLevel:
-		levelColor = red
-	default:
-		levelColor = blue
+// SetLevelColor sets a color to a log level in the ColorMap
+func (f *TextFormatter) SetLevelColor(l Level, c uint8) {
+	if f.colorMap == nil {
```

Comment:
When will colorMap be nil here? It should always be set during `init`. So this condition should be removable.

Reply:
The `init(entry *Entry)` call is currently implemented in the `Format(entry *Entry)` method. If you want to set a new color **before** a log entry call you need to set the color map. 
You can check [here](https://github.com/msiebeneicher/logrus/blob/feature/add_color_setting_support_in_text_formatter/example_basic_test.go) the updated example.

Alternative you can provide a func NewTextFormatter() function.